providers/explainshell/explainshell.go

```diff
@@ -0,0 +1,30 @@
+package explainshell
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/zquestz/s/providers"
+)
+
+// strings.Replace will apply for all matches when n < 0
+const REPLACE_ALL int = -1
+const PROVIDER_NAME = "explainshell"
+const PROVIDER_URL = "http://explainshell.com"
+
+func init() {
+	providers.AddProvider(PROVIDER_NAME, &Provider{})
+}
+
+// Provider merely implements the Provider interface
+type Provider struct{}
+
+// BuildURI generates a search URL for Explain Shell
+func (p *Provider) BuildURI(q string) string {
+	searchParam := strings.Replace(q, " ", "+", REPLACE_ALL)
+	return fmt.Sprintf("%s/explain?cmd=%s", PROVIDER_URL, searchParam)
+}
+
+func (p *Provider) Tags() []string {
+	return []string{"shell", "tech"}
```

Comment:
Should be in "code" tag. Try to keep to existing tags unless there is a good argument to be made for a new one.